code/game/objects/items/robot/robot_items.dm

```diff
@@ -34,3 +35,190 @@
+								"<span class='warning'>You hug [M] firmly to make \him feel better! [M] looks uncomfortable...</span>")
+					if(M.resting)
+						M.resting = FALSE
+						M.update_canmove()
+				else
+					user.visible_message("<span class='warning'>[user] bops [M] on the head!</span>", \
+							"<span class='warning'>You bop [M] on the head!</span>")
+				playsound(loc, 'sound/weapons/tap.ogg', 50, 1, -1)
+		if(2)
+			if(!scooldown)
+				if(M.health >= 0)
+					if(ishuman(M))
+						M.electrocute_act(5, "[user]", safety = 1)
+						user.visible_message("<span class='userdanger'>[user] electrocutes [M] with their touch!</span>", \
+							"<span class='danger'>You electrocute [M] with your touch!</span>")
+						M.update_canmove()
+					else
+						if(!isrobot(M))
+							M.adjustFireLoss(10)
+							user.visible_message("<span class='userdanger'>[user] shocks [M]!</span>", \
+								"<span class='danger'>You shock [M]!</span>")
+						else
+							user.visible_message("<span class='userdanger'>[user] shocks [M]. It does not seem to have an effect</span>", \
+								"<span class='danger'>You shock [M] to no effect.</span>")
+					playsound(loc, 'sound/effects/sparks2.ogg', 50, 1, -1)
+					user.cell.charge -= 500
+					scooldown = TRUE
+					spawn(20)
+						scooldown = FALSE
+		if(3)
+			if(!ccooldown)
+				if(M.health >= 0)
+					if(ishuman(M))
+						user.visible_message("<span class='userdanger'>[user] crushes [M] in their grip!</span>", \
+							"<span class='danger'>You crush [M] in your grip!</span>")
+					else
+						user.visible_message("<span class='userdanger'>[user] crushes [M]!</span>", \
+								"<span class='danger'>You crush [M]!</span>")
+					playsound(loc, 'sound/weapons/smash.ogg', 50, 1, -1)
+					M.adjustBruteLoss(15)
+					user.cell.charge -= 300
+					ccooldown = TRUE
+					spawn(10)
+						ccooldown = FALSE
+
+/obj/item/borg/cyborghug/peacekeeper
+	shockallowed = TRUE
+
+/obj/item/device/harmalarm
+	name = "Sonic Harm Prevention Tool"
+	desc = "Releases a harmless blast that confuses most organics. For when the harm is JUST TOO MUCH"
+	icon_state = "megaphone"
+	var/cooldown = 0
+	var/emagged = FALSE
+
+/obj/item/device/harmalarm/emag_act(mob/user)
+	emagged = !emagged
+	if(emagged)
+		to_chat(user, "<font color='red'>You short out the safeties on the [src]!</font>")
+	else
+		to_chat(user, "<font color='red'>You reset the safeties on the [src]!</font>")
+
+/obj/item/device/harmalarm/attack_self(mob/user)
+	var/safety = !emagged
+	if(cooldown > world.time)
+		to_chat(user, "<font color='red'>The device is still recharging!</font>")
+		return
+
+	if(isrobot(user))
+		var/mob/living/silicon/robot/R = user
+		if(R.cell.charge < 1200)
+			to_chat(user, "<font color='red'>You don't have enough charge to do this!</font>")
+			return
+		R.cell.charge -= 1000
+		if(R.emagged)
+			safety = FALSE
+
+	if(safety == TRUE)

```

Comment:
Whyyyy. Just check `if(safety)` and then completely kill the `if(safety == FALSE)` block, because it's redundant- just making it inline with the proc. `if(safety)` always returns before that point, so logically anything after the bottom of this block means that safety is not true.